### DISCUSSION TITLE
Limit of 499 entries for .include

### DISCUSSION THREAD

**Discussion by jbonigomes** [jsforce/jsforce#1442] [Q&A]
Does JSForce limit the results coming from a relationship with .include()? I have a custom relationship in my Account table, but my inner query does not return more than 499 records:

```Javascript
  conn
    .sobject('Account')
    .select('the,fields,I,need')
    .include('Account_Contacts__r')
      .select('the,other,fields,I,need')
      .end()
    .execute({ autoFetch: true, maxFetch: 1000000 })
    .on('record', (res) => {
      // process the records
      // ...
    });
```

I have also tried adding a limit to it, with no success:

```Javascript
  conn
    .sobject('Account')
    .select('the,fields,I,need')
    .include('Account_Contacts__r')
      .select('the,other,fields,I,need')
      .limit(1000000)
      .end()
    .execute({ autoFetch: true, maxFetch: 1000000 })
    .on('record', (res) => {
      // process the records
      // ...
    });
```

**Comment by apashetnykh** [comment]
Do you have solution for fix or workaround it?

**Comment by paustint** [comment]
I am fairly certain that SFDC paginates the data for subqueries, so it may not return all records in the initial query.  Providing a limit will not impact how many records are returned with each set of query results.

The only thing I can think to do would be to use the queryLocator (returned as part of the subquery) to fetch the next set of records and continue until they are done.  Unfortunately since this is a subquery, you will have to do it for each of parent records that have remaining child records that were not returned.

You might want to re-think your query strategy if you have large numbers of child records (e.x. query child records and include parent fields and in javascript reduce the data back into the parent child structure)
